examples/vision/image_classification_efficientnet_fine_tuning.py

```diff
@@ -0,0 +1,423 @@
+EfficientNetB0, image size is 224.
+"""
+
+# IMG_SIZE is determined by EfficientNet model choice
+IMG_SIZE = 224
+
+"""
+### prepare
+"""
+
+"""shell
+!pip install --quiet tensorflow==2.3.0rc0
+!pip install --quiet cloud-tpu-client
+"""
+import tensorflow as tf
+
+try:
+    from cloud_tpu_client import Client
+
+    c = Client()
+    c.configure_tpu_version(tf.__version__, restart_type="always")
+    tpu = tf.distribute.cluster_resolver.TPUClusterResolver()  # TPU detection
+    print("Running on TPU ", tpu.cluster_spec().as_dict()["worker"])
+    tf.config.experimental_connect_to_cluster(tpu)
+    tf.tpu.experimental.initialize_tpu_system(tpu)
+    strategy = tf.distribute.experimental.TPUStrategy(tpu)
+except ValueError:
+    print("Not connected to a TPU runtime. Using CPU/GPU strategy")
+    strategy = tf.distribute.MirroredStrategy()
+
+
+"""
+Below is example code for loading data.
+To see sensible result, you need to load entire dataset and adjust epochs for
+training; but you may truncate data for a quick verification of the workflow.
+Expect the notebook to run at least an hour for GPU, while much faster on TPU if
+using hosted Colab session.
+"""
+
+from tensorflow import keras
+from tensorflow.keras.datasets import cifar100
+from tensorflow.keras.utils import to_categorical
+
+batch_size = 64
+
+(x_train, y_train), (x_test, y_test) = cifar100.load_data()
+NUM_CLASSES = 100
+
+x_train = tf.cast(x_train, tf.int32)
+x_test = tf.cast(x_test, tf.int32)
+
+truncate_data = False  # @param {type: "boolean"}
+if truncate_data:
+    x_train = x_train[0:5000]
+    y_train = y_train[0:5000]
+    x_test = x_test[0:1000]
+    y_test = y_test[0:1000]
+
+
+# one-hot / categorical
+y_train = to_categorical(y_train, NUM_CLASSES)
+y_test = to_categorical(y_test, NUM_CLASSES)
+
+ds_train = tf.data.Dataset.from_tensor_slices((x_train, y_train))
+ds_train = ds_train.cache()
+ds_train = ds_train.batch(batch_size=batch_size, drop_remainder=True)
+ds_train = ds_train.prefetch(tf.data.experimental.AUTOTUNE)
+
+ds_test = tf.data.Dataset.from_tensor_slices((x_test, y_test))
+ds_test = ds_test.batch(batch_size=batch_size, drop_remainder=True)
+
+"""
+### training from scratch
+To build model that use EfficientNetB0 with 100 classes that is initiated from scratch:
+
+Note: to better see validation peeling off from training accuracy, run ~20 epochs.
+"""
+
+from tensorflow.keras.applications import EfficientNetB0
+from tensorflow.keras.layers.experimental.preprocessing import (
+    Resizing,
+    RandomFlip,
+    RandomContrast,
+    # RandomHeight,
+)
+from tensorflow.keras.optimizers import SGD
+
+with strategy.scope():
+    inputs = keras.layers.Input(shape=(32, 32, 3))
+    x = inputs
+
+    x = RandomFlip()(x)
+    x = RandomContrast(0.1)(x)
+    # x = RandomHeight(0.1)(x)
+    x = Resizing(IMG_SIZE, IMG_SIZE, interpolation="bilinear")(x)
+
+    x = EfficientNetB0(include_top=True, weights=None, classes=100)(x)
```

Comment:
Since you are using a constant for number of classes above
```suggestion
    x = EfficientNetB0(include_top=True, weights=None, classes=NUM_CLASSES)(x)
```